numpyro/infer/mcmc.py

```diff
@@ -382,11 +383,76 @@ def _compile(self, rng_key, *args, extra_fields=(), init_params=None, **kwargs):
         except TypeError:
             pass
 
+    @property
+    def init_state(self):
+        """
+        The initial state to run MCMC. If this attribute is not None,
+        :meth:`run` will skip calling `self.sampler.init(...)` method to
+        obtain the initial state.
+
+        .. warning:: If you want to set a value for this attribute, then making sure
+            that some fields such as `HMCState.i`, `HMCState.mean_accept_prob` are
+            reseted to 0. In addition, if the initial state has invalid fields such as
+            `HMCState.potential_energy` (because e.g. `data` of the new `run` has been
+            changed), the sampler might get trouble during some initial MCMC steps.
+            To resolve that, we can set `potential_energy` to a high value (says `1e38`),
+            so that it is more likely to accept a new proposal (which will have a correct
+            `potential_energy` field) in the first MCMC step.
+            For example
+
+            .. code-block:: python
+
+                mcmc = MCMC(NUTS(model), 100, 100)
+                mcmc.run(random.PRNGKey(0), data0)
+                samples_for_data0 = mcmc.get_samples()
+                mcmc.init_state = mcmc.last_state._replace(
+                    i=0, potential_energy=1e38, mean_accept_prob=0.)
+                mcmc.run(random.PRNGKey(1), data1)
+                samples_for_data1 = mcmc.get_samples()
+
+        """
+        return self._init_state
+
+    @init_state.setter
+    def init_state(self, state):
+        self._init_state = state
+
+    @property
+    def warmup_state(self):
+        """
+        The state before sampling phase. If this attribute is not None,
```

Comment:
nit:
        The state before the sampling phase. If this attribute is not None,
        :meth:`run` will skip the warmup phase and start with the state
        specified in this attribute.